### PR TITLE
Cnft1 2438 2439 fields form lab report

### PR DESCRIPTION
**Pull request by hclarkEnq**
## Description

Form Criteria UI for Lab Reports 

## Tickets

* [CNFT1-2438](https://cdc-nbs.atlassian.net/browse/CNFT1-2438)
* [CNFT1-2439](https://cdc-nbs.atlassian.net/browse/CNFT1-2439)

## Checklist before requesting a review
- [ ] PR focuses on a single story
- [ ] Code has been fully tested to meet acceptance criteria
- [ ] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [ ] All new functions/classes/components reasonably small
- [ ] Functions/classes/components focused on one responsibility
- [ ] Code easy to understand and modify (clarity over concise/clever)
- [ ] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [ ] PR does not contain hardcoded values (Uses constants)
- [ ] All code is covered by unit or feature tests

[CNFT1-2438]: https://cdc-nbs.atlassian.net/browse/CNFT1-2438?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CNFT1-2439]: https://cdc-nbs.atlassian.net/browse/CNFT1-2439?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/modernization-ui/src/apps/search/laboratory-report/labReportFormTypes.ts [hunk 1]
```diff
@@ -23,9 +24,11 @@ type General = {
     enteredBy?: Selectable[];
     entryMethods?: Selectable[];
     eventDate?: EventDate;
+    eventId?: LabReportEventId | null;
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/search/laboratory-report/labReportFormTypes.ts:27]: issue: Let's try to reduce the usage of the generated GraphQL types as much as possible.  The `identification` property is the same as `eventId`, the transformer already converts to it

File: apps/modernization-ui/src/apps/search/laboratory-report/labReportFormTypes.ts [hunk 2]
```diff
@@ -23,9 +24,11 @@ type General = {
...
     identification?: Identification;
     eventStatus?: Selectable[];
     jurisdictions?: Selectable[];
+    lastUpdatedBy?: Selectable;
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/search/laboratory-report/labReportFormTypes.ts:31]: issue: use `updatedBy`

File: apps/modernization-ui/src/apps/search/laboratory-report/labReportFormTypes.ts [hunk 3]
```diff
@@ -34,7 +37,8 @@ type General = {
     pregnancyStatus?: Selectable;
     processingStatus?: Selectable[];
     programAreas?: Selectable[];
-};
+    providerSearch?: LabReportProviderSearch | null;
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/search/laboratory-report/labReportFormTypes.ts:40]: issue:  use the properties that were added for the specific facility/ provider.  You can display the appropriate one when the type is selected.

File: apps/modernization-ui/src/apps/search/laboratory-report/CriteriaFields.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+};
+
+const CriteriaFieldsWithForm = () => {
+    const form = useForm<LabReportFilterEntry>({ defaultValues: {} });
+    return <CriteriaFields form={form} />;
+};
+
+describe('LabReportCriteriaFields component', () => {
+    it('should render ResultedTestsAutocomplete with correct props', async () => {
+        const { getByTestId, getByPlaceholderText } = render(<LabReportCriteriaFieldsWithForm />);
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/search/laboratory-report/CriteriaFields.spec.tsx:41]: suggestion: The `getByTestId` should be used as a last resort [getByRole](https://testing-library.com/docs/queries/byrole) can be used here

File: apps/modernization-ui/src/apps/search/layout/SearchLayout.tsx [hunk 5]
```diff
@@ -10,7 +10,7 @@ import { SearchNavigation } from './navigation/SearchNavigation';
 
 type Renderer = () => ReactNode;
 
-type Props = {
+export type Props = {
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/search/layout/SearchLayout.tsx:13]: question (non-blocking): What needed this type to be exported?

File: apps/modernization-ui/src/design-system/select/single/SingleSelect.tsx [hunk 6]
```diff
@@ -25,7 +25,7 @@ type Props = {
     required?: boolean;
 } & Omit<JSX.IntrinsicElements['select'], 'defaultValue' | 'onChange' | 'value'>;
 
-const SinlgeSelect = ({
+const SingleSelect = ({
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/design-system/select/single/SingleSelect.tsx:28]: praise:  Thanks for fixing the spelling!

File: apps/modernization-ui/src/apps/search/layout/SearchLayout.tsx (hunk 5 above)
Comment (hclarkEnq) [reply apps/modernization-ui/src/apps/search/layout/SearchLayout.tsx:13]: I was using it somewhere before but with the recent changes I removed it so I am reverting this back

**Review by adamloup-enquizit — CHANGES_REQUESTED** [review]

**Review by jlee-enquizit — APPROVED** [review]

File: apps/modernization-ui/src/apps/search/laboratory-report/labReportFormTypes.ts [hunk 7]
```diff
@@ -23,6 +23,8 @@ type General = {
     enteredBy?: Selectable[];
     entryMethods?: Selectable[];
     eventDate?: EventDate;
+    eventId?: Identification['value'];
+    eventType?: Identification['type'];
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/search/laboratory-report/labReportFormTypes.ts:27]: issue:  Please use the `identification` property.  The transformer is already set up to convert from the `Identification` type to the `LabReportEventId` type expected by the API.

File: apps/modernization-ui/src/apps/search/laboratory-report/labReportFormTypes.ts [hunk 8]
```diff
@@ -34,7 +36,8 @@ type General = {
     pregnancyStatus?: Selectable;
     processingStatus?: Selectable[];
     programAreas?: Selectable[];
-};
+    providerType?: Selectable;
+} & ReportCriteria;
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/search/laboratory-report/labReportFormTypes.ts:40]: nit: This is redundant, the `LabReportFiterEntry` is composed of `General` and `ReportCriteria`.

**Review by jlee-enquizit — DISMISSED** [review]

**Review by adamloup-enquizit — APPROVED** [review]
Nice! 👍

**Review by jlee-enquizit — APPROVED** [review]

**Pull request merged**